### PR TITLE
Add support for rand and has any experssions

### PR DESCRIPTION
**Pull request by ofrikleinfeld**

### CODE REVIEW

File: pykusto/_src/functions.py [hunk 1]
```diff
@@ -1208,7 +1208,7 @@ def dcount(expr: ExpressionType, accuracy: NumberType = None) -> _NumberAggregat
         return _NumberAggregationExpression(KQL(f'dcount({_to_kql(expr)})' if accuracy is None else f'dcount({_to_kql(expr)}, {_to_kql(accuracy)})'))
 
     @staticmethod
-    def dcount_if(expr: ExpressionType, predicate: BooleanType, accuracy: NumberType = 0) -> _NumberAggregationExpression:
+    def dcount_if(expr: ExpressionType, predicate: BooleanType, accuracy: NumberType = 1) -> _NumberAggregationExpression:
```
Comment (ofrikleinfeld) [inline pykusto/_src/functions.py:1211]: Please see [https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/dcountif-aggfunction](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/dcountif-aggfunction)
The default accuracy is 1, which is more accurate than 0.
I pyKusto should not change the default behavior expected by kusto users working with the web/desktop client

**Review by amos-rimon — APPROVED** [review]

File: pykusto/_src/expressions.py [hunk 2]
```diff
@@ -523,6 +523,13 @@ def has_not(self, exp: StringType, case_sensitive: bool = False) -> '_BooleanExp
             f'{self.as_subexpression()} {"!has_cs" if case_sensitive else "!has"} {_to_kql(exp, True)}'
         ))
 
+    def has_any(self, other: ArrayType) -> '_BooleanExpression':
+        """
+        https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/has-anyoperator
+        """
+        assert isinstance(other, (List, Tuple)), "Compared array must be a list of tabular, scalar, or literal expressions"
```
Comment (ymost) [inline pykusto/_src/expressions.py:530]: Only a literal list is supported, so why is the declared type `ArrayType`? It should be `Union[List, Tuple]` or something similar. 
Even better, we can use `*other: str`, which indicates more clearly that a list of strings is needed.
Also, this implementation does not support a tabular expression, so let's state that in the documentation.

File: pykusto/_src/functions.py [hunk 3]
```diff
@@ -645,11 +645,11 @@ def pow(expr1: NumberType, expr2: NumberType) -> _NumberExpression:
...
     # def radians(self): return
-    #
-    #
-    # def rand(self): return
-    #
-    #
+
+    @staticmethod
+    def rand(n: NumberType = None):
+        return _NumberExpression(KQL("rand()") if n is None else f'rand({_to_kql(n)})')
```
Comment (ymost) [inline pykusto/_src/functions.py:651]: Add a link to the Kusto documentation

File: pykusto/_src/expressions.py (hunk 2 above)
Comment (ofrikleinfeld) [reply pykusto/_src/expressions.py:530]: I changed the argument type as advised, and added a matching comment in the operator documentation. I'm not in favor of the `*other: str` option and prefer to stay with the explicit iterable type.

**Pull request merged**